src/os/bluestore/BlueFS.cc

```diff
@@ -1414,6 +1414,61 @@ int BlueFS::_replay(bool noop, bool to_stdout)
 	  }
         }
 	break;
+      case bluefs_transaction_t::OP_FILE_UPDATE_INC:
+	{
+	  bluefs_fnode_delta_t delta;
+	  decode(delta, p);
+	  dout(20) << __func__ << " 0x" << std::hex << pos << std::dec
+	    << ":  op_file_update_inc " << " " << delta << " " << dendl;
+	  if (unlikely(to_stdout)) {
+	    std::cout << " 0x" << std::hex << pos << std::dec
+	      << ":  op_file_update_inc " << " " << delta << std::endl;
+	  }
+	  if (!noop) {
+	    FileRef f = _get_file(delta.ino);
+	    bluefs_fnode_t& fnode = f->fnode;
+	    if (delta.offset != fnode.allocated) {
+	      derr << __func__ << " invalid op_file_update_inc, new extents miss end of file"
+		   << " fnode=" << fnode
+		   << " delta=" << delta
+		   << dendl;
+	      ceph_assert(delta.offset == fnode.allocated);
```

Comment:
```
2021-08-20T09:18:19.015+0000 7f5ac7cc9180  1 bluefs _init_alloc new, id 1, allocator name bluefs-db, allocator type stupid, capacity 0x88000000, block size 0x1000
2021-08-20T09:18:19.031+0000 7f5ac7cc9180 -1 bluefs _replay invalid op_file_update_inc, new extents miss end of file fnode=file(ino 1 size 0xd000 mtime 0.000000 allocated 14000 alloc_commit 14000 extents [1:0x30d2000~4000,1:0x30c2000~10000]) delta=delta(ino 1 size 0xd000 mtime 0.000000 offset 0 extents [1:0x30d2000~4000,1:0x30c2000~10000,1:0x30da000~10000])
../src/os/bluestore/BlueFS.cc: In function 'int BlueFS::_replay(bool, bool)' thread 7f5ac7cc9180 time 2021-08-20T09:18:19.035017+0000
../src/os/bluestore/BlueFS.cc: 1435: FAILED ceph_assert(delta.offset == fnode.allocated)
 ceph version Development (no_version) quincy (dev)
 1: (ceph::__ceph_assert_fail(char const*, char const*, int, char const*)+0x1b4) [0x7f5ac9bd09b4]
 2: /home/jenkins-build/build/workspace/ceph-pull-requests/build/lib/libceph-common.so.2(+0x13d47ff) [0x7f5ac9bd07ff]
 3: (BlueFS::_replay(bool, bool)+0x566f) [0x62ff6f]
 4: (BlueFS::mount()+0x499) [0x629389]
 5: (BlueFS_test_replay_growth_Test::TestBody()+0xd57) [0x5af117]
 6: (void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*)+0x7b) [0x615f4b]
 7: (void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*)+0x6a) [0x5fa88a]
 8: (testing::Test::Run()+0xc3) [0x5d8243]
 9: (testing::TestInfo::Run()+0xda) [0x5d8f8a]
 10: (testing::TestSuite::Run()+0xfb) [0x5d97ab]
 11: (testing::internal::UnitTestImpl::RunAllTests()+0x442) [0x5e6ec2]
 12: (bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*)+0x7b) [0x618f6b]
 13: (bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*)+0x6a) [0x5fd2aa]
 14: (testing::UnitTest::Run()+0xcb) [0x5e6a2b]
 15: (RUN_ALL_TESTS()+0x11) [0x5b4841]
 16: main()
 17: __libc_start_main()
 18: _start()
```

https://jenkins.ceph.com/job/ceph-pull-requests/81456/consoleFull#-108728127277933967-90d1-4877-8d60-89cb08ef4eb1

Reply:
@aclamk Seeing tests fail with this in http://pulpito.front.sepia.ceph.com/yuriw-2021-09-23_01:04:10-rados-wip-yuri6-testing-2021-09-22-1530-distro-basic-smithi/

```
2021-09-23T02:21:55.942 INFO:teuthology.orchestra.run.smithi085.stderr:/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/17.0.0-7843-g2b4d84fd/rpm/el8/BUILD/ceph-17.0.0-7843-g2b4d84fd/src/os/bluestore/BlueFS.cc: In function 'int BlueFS::_replay(bool, bool)' thread 7fd98a9e6ec0 time 2021-09-23T02:21:55.942917+0000
2021-09-23T02:21:55.943 INFO:teuthology.orchestra.run.smithi085.stderr:/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/17.0.0-7843-g2b4d84fd/rpm/el8/BUILD/ceph-17.0.0-7843-g2b4d84fd/src/os/bluestore/BlueFS.cc: 1447: FAILED ceph_assert(delta.offset == fnode.allocated)
2021-09-23T02:21:55.944 DEBUG:teuthology.orchestra.run.smithi085:> sudo adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage timeout 30 ceph --cluster ceph --admin-daemon /var/run/ceph/ceph-osd.3.asok dump_ops_in_flight
2021-09-23T02:21:55.946 INFO:teuthology.orchestra.run.smithi085.stderr: ceph version 17.0.0-7843-g2b4d84fd (2b4d84fd3bb01971f260800980324d2f624fea94) quincy (dev)
2021-09-23T02:21:55.946 INFO:teuthology.orchestra.run.smithi085.stderr: 1: (ceph::__ceph_assert_fail(char const*, char const*, int, char const*)+0x152) [0x7fd98896cb7e]
2021-09-23T02:21:55.947 INFO:teuthology.orchestra.run.smithi085.stderr: 2: /usr/lib64/ceph/libceph-common.so.2(+0x284d9f) [0x7fd98896cd9f]
2021-09-23T02:21:55.947 INFO:teuthology.orchestra.run.smithi085.stderr: 3: (BlueFS::_replay(bool, bool)+0x3e34) [0x55ae9a885e94]
2021-09-23T02:21:55.948 INFO:teuthology.orchestra.run.smithi085.stderr: 4: (BlueFS::mount()+0x120) [0x55ae9a887ee0]
2021-09-23T02:21:55.948 INFO:teuthology.orchestra.run.smithi085.stderr: 5: (BlueStore::_open_bluefs(bool, bool)+0x94) [0x55ae9a7689f4]
2021-09-23T02:21:55.948 INFO:teuthology.orchestra.run.smithi085.stderr: 6: (BlueStore::_prepare_db_environment(bool, bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)+0x6d9) [0x55ae9a769b49]
2021-09-23T02:21:55.949 INFO:teuthology.orchestra.run.smithi085.stderr: 7: (BlueStore::_open_db(bool, bool, bool)+0x15c) [0x55ae9a76ae6c]
2021-09-23T02:21:55.949 INFO:teuthology.orchestra.run.smithi085.stderr: 8: (BlueStore::_open_db_and_around(bool, bool)+0x27d) [0x55ae9a7e27cd]
2021-09-23T02:21:55.950 INFO:teuthology.orchestra.run.smithi085.stderr: 9: (BlueStore::_fsck(BlueStore::FSCKDepth, bool)+0x5b) [0x55ae9a7e4ebb]
2021-09-23T02:21:55.950 INFO:teuthology.orchestra.run.smithi085.stderr: 10: (BlueStore::_mount()+0x7b) [0x55ae9a7e548b]
2021-09-23T02:21:55.951 INFO:teuthology.orchestra.run.smithi085.stderr: 11: main()
2021-09-23T02:21:55.951 INFO:teuthology.orchestra.run.smithi085.stderr: 12: __libc_start_main()
2021-09-23T02:21:55.951 INFO:teuthology.orchestra.run.smithi085.stderr: 13: _start()
```